Compentio.SourceMapper.Generator/Generators/CodeSourceGenerator.cs

```diff
@@ -54,11 +58,23 @@ internal void GenerateDependencyInjectionExtensions(GeneratorExecutionContext co
             context.AddSource($"{_sourcesMetadata.DependencyInjection.DependencyInjectionClassName}.cs", SourceText.From(result.GeneratedCode, Encoding.UTF8));
         }
 
+        private void AddNoDependencyInjectionDiagnostic(GeneratorExecutionContext context)
+        {
+            _diagnostics.Add(new DiagnosticsInfo()
+            {
+                DiagnosticDescriptor = SourceMapperDescriptors.DependencyInjectionNotUsed,
+                Exception = new ExtException(AppStrings.DependencyInjectionNotUsedException, context.Compilation.AssemblyName)
```

Comment:
Or maybe to report diagnostics without throwing exception just using `Location`

Reply:
Exception is only added to diagnostics, not throwing - is should be showing during compilation as other diagnostics messages.